charmdir.go

```diff
@@ -316,3 +336,24 @@ func checkFileType(path string, mode os.FileMode) error {
 	}
 	return fmt.Errorf(e, path)
 }
+
+// MaybeCreateVersionFile creates/overwrite charm version file.
+func MaybeGenerateVersionString(path string) (string, error) {
+	var cmdArgs []string
+	// Verify that it is revision control directory.
+	if _, err := os.Stat(filepath.Join(path, ".hg")); err == nil {
+		cmdArgs = []string{"hg", "id", "-n"}
+	} else if _, err = os.Stat(filepath.Join(path, ".bzr")); err == nil {
+		cmdArgs = []string{"bzr", "version-info"}
+	} else if _, err = os.Stat(filepath.Join(path, ".git")); err == nil {
+		cmdArgs = []string{"git", "describe", "--dirty"}
+	} else {
+		logger.Debugf("charm is not in revision control directory")
+		// Return empty string.
+		return "", nil
+	}
+
+	//return utils.RunCommand(cmdArgs[0], cmdArgs[1:]...)
+	str, err := utils.RunCommand(cmdArgs[0], cmdArgs[1:]...)
```

Comment:
the bit that was commented out is all we need

Reply:
this was debug

Reply:
reverting and uncomment the original code